# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382026284 25200
#      Thu Oct 17 09:11:24 2013 -0700
# Node ID 36917b3db6b2654ad6f824143cd9a2a383c9510f
# Parent  ddfc1600a31185f393d418328b3562c483a5300f
orient.m: Setting 'landscape' orientation also sets plot to take up full page.

* scripts/plot/appearance/orient.m: Check for 'landscape' orientation and
set paperposition to be entire page minus 0.25" margins.

diff --git a/scripts/plot/appearance/orient.m b/scripts/plot/appearance/orient.m
--- a/scripts/plot/appearance/orient.m
+++ b/scripts/plot/appearance/orient.m
@@ -60,16 +60,21 @@ function retval = orient (varargin)
         ## FIXME: with the proper listeners in place there won't be a need to
         ##        set the papersize and paperpostion here.
         papersize = get (cf, "papersize");
         paperposition = get (cf, "paperposition");
         set (cf, "paperorientation", orientation);
         set (cf, "papersize", papersize([2, 1]));
         set (cf, "paperposition", paperposition([2, 1, 4, 3]));
       endif
+      ## landscape also sets the plot to occupy the entire page
+      if (strcmpi (orientation, "landscape"))
+        papersize = get (cf, "papersize");
+        set (cf, "paperposition", [0.25, 0.25, (papersize - 0.5)]);
+      endif
     elseif (strcmpi (varargin{1}, 'tall'))
       orient ("portrait");
       papersize = get (cf, "papersize");
       set (cf, "paperposition", [0.25, 0.25, (papersize - 0.5)]);
     else
       error ("orient: unknown ORIENTATION");
     endif
   else
